tests/unit/test_logger.py

```diff
@@ -0,0 +1,105 @@
+import pytest
+
+from guidellm import LoggerConfig, configure_logger, logger
+
+
+@pytest.fixture(autouse=True)
+def reset_logger():
+    # Ensure logger is reset before each test
+    logger.remove()
+    yield
+    logger.remove()
```

Comment:
I believe that we could keep in in a `tests/conftest.py` so this fixture is available for the whole project. What do you think?

Reply:
Another thing that we could bring up is using of the `pytest_configure` reserved hook that could include the logger configuration.

```py
# /tests/conftest.py
def pytest_configure() -> None:
     logger.remove()
```

Reply:
Good idea, I'm going to push this off for now to another PR to combine with the pydantic env management